### PR TITLE
Bring back in KVS migration

### PR DESCRIPTION
**Pull request by tmgordeeva**

### CODE REVIEW

**Comment by tmgordeeva** [comment]
@clockfort @jboreiko this is the external part of the internal refactoring you've seen.

File: atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/KvsMigrationCommand.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        toServices.getTransactionService().putUnlessExists(migrationTimestampSupplier.get(), toServices.getTimestampService().getFreshTimestamp());
+        return new KeyValueServiceMigrator(
+                CHECKPOINT_NAMESPACE,
+                fromServices.getTransactionManager(),
+                toServices.getTransactionManager(),
+                fromServices.getKeyValueService(),
+                toServices.getKeyValueService(),
+                migrationTimestampSupplier,
+                threads,
+                256,
```
Comment (clockfort) [inline atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/KvsMigrationCommand.java]: named constant would be preferable

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                    return null;
+                }
+            });
+            futures.add(future);
+        }
+
+        futures.forEach(future -> Futures.getUnchecked(future));
+    }
+
+    private void validateTable(final TableReference table) {
```
Comment (clockfort) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java]: could be de-duped

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+    private void validateEquality(Map<Cell, byte[]> cells1, Map<Cell, byte[]> cells2) {
+        Set<Cell> ks1 = cells1.keySet();
+        Set<Cell> ks2 = cells2.keySet();
+        if (!ks1.equals(ks2)) {
+            Validate.isTrue(false, "Cells not equal. Expected: " + ks1 + ". Actual: " + ks2);
+        }
+        for (Cell c : ks1) {
+            if (!Arrays.equals(cells1.get(c), cells2.get(c))) {
+                Validate.isTrue(false, "Value not equal in " + c);
```
Comment (clockfort) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java]: who writes things this way

**Comment by clockfort** [comment]
looks fine, feel free to merge if you don't care about the comments

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java (hunk 3 above)
Comment (tmgordeeva) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java]: This was probably this way to avoid string concats.

Comment (schlosna) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java]: Following would work:

```
private void validateEquality(Map<Cell, byte[]> cells1, Map<Cell, byte[]> cells2) {
    Set<Cell> ks1 = cells1.keySet();
    Set<Cell> ks2 = cells2.keySet();
    Preconditions.checkArgument(ks1.equals(ks2),
            "Cells not equal. Expected: %s. Actual: %s", ks1, ks2);
    for (Cell c : ks1) {
        Preconditions.checkArgument(Arrays.equals(cells1.get(c), cells2.get(c)),
                "Values not equal for cell %s", c);
    }
}
```

**Comment by jlz27** [comment]
@tmgordeeva what is the behavior of this cli if the destination kvs already has the same tables?

**Pull request merged**